pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -73,57 +85,94 @@ func (w *HashAggFinalWorker) initBInMap() {
 	}
 }
 
-func (w *HashAggFinalWorker) consumeIntermData(sctx sessionctx.Context) (err error) {
-	for {
-		waitStart := time.Now()
-		input, ok := w.getPartialInput()
-		if w.stats != nil {
-			w.stats.WaitTime += int64(time.Since(waitStart))
+func (w *HashAggFinalWorker) getInputData() (*aggfuncs.AggPartialResultMapper, bool, bool) {
+	waitStart := time.Now()
+	defer w.increaseWaitTime(waitStart)
+	if w.isSpilledTriggered {
+		// Only one partition is restored each time, so we should execute in this loop only once.
```

Comment:
I don't understand the comment, I don't see any loop here.

Reply:
> I don't understand the comment, I don't see any loop here.

The loop is outside of this function, it's meaningless now, I will delete it.